### PR TITLE
Rename operator mapping map() to apply()

### PR DESCRIPTION
**Pull request by uranusjr**
According to the design discussion, we’re going to use `map()` for something else that matches more closely to the FP `map` concept.

Do we also want to rename `mapped_kwargs` to something else? (to what?)

### CODE REVIEW

**Review by ashb — APPROVED** [review]

File: airflow/decorators/task_group.py [hunk 1]
```diff
@@ -106,12 +107,16 @@ def _make_task_group(self, **kwargs) -> MappedTaskGroup:
         return tg
 
     def partial(self, **kwargs) -> "MappedTaskGroupDecorator[R]":
-        if self.partial_kwargs:
-            raise RuntimeError("Already a partial task group")
+        duplicated_keys = [k for k in kwargs if k in self.partial_kwargs]
+        if len(duplicated_keys):
```
Comment (dstandish) [inline airflow/decorators/task_group.py]:
```suggestion
        if len(duplicated_keys) == 1:
```

File: airflow/decorators/task_group.py [hunk 2]
```diff
@@ -106,12 +107,16 @@ def _make_task_group(self, **kwargs) -> MappedTaskGroup:
...
+            raise ValueError(f"Cannot overwrite partial argument: {duplicated_keys[0]!r}")
+        elif duplicated_keys:
+            joined = ", ".join(repr(k) for k in duplicated_keys)
+            raise ValueError(f"Cannot overwrite partial arguments: {joined}")
```
Comment (dstandish) [inline airflow/decorators/task_group.py]: or alternatively... 

```suggestion
duplicated_keys = [k for k in kwargs if k in self.partial_kwargs]
if duplicated_keys:
    raise ValueError(f"Cannot overwrite partial arguments: {duplicated_keys!r}")
```

personally i wouldn't sweat the grammar but different strokes :)

**Review by dstandish — APPROVED** [review]
i think i caught one small thing with the `len` comparison but otherwise looks good

File: airflow/decorators/task_group.py (hunk 1 above)
Comment (uranusjr) [reply airflow/decorators/task_group.py]: Ah, of course, thanks!

**Pull request merged**